content/en/docs/application-store/built-in-apps/harbor-app.md

```diff
@@ -6,5 +6,101 @@ description: 'How to use built-in Harbor registry'
+
+## Hands-on Lab
+
+### Common steps
+
+1. To begin with, you deploy Harbor from the App Store. Please make sure you are landing on the **Overview** page of the project `demo`.
+
+![overview_of_demo_project](/images/docs/appstore/harbor/overview_of_demo_project.png)
+
+2. List your deployed apps. If it is empty, click the button `Deploy New Application`.
+
+![new_app_deploy](/images/docs/appstore/harbor/new_app_deploy.png)
+
+3. Choose harbor template `From App Store`.
+
+![choose_app_from_store](/images/docs/appstore/harbor/choose_app_from_store.png)
+
+4. Find harbor in App Store.
+
+![overview_of_apps](/images/docs/appstore/harbor/overview_of_apps.png)
+
+5. Choose harbor **version** and **deployment location**, then click `Next`.
+
+![deploy_set_of_harbor](/images/docs/appstore/harbor/deploy_set_of_harbor.png)
+
+6. Config harbor yaml, then click `Deploy`. There was an example yaml in section **FAQ**.
+
+![config_of_harbor_deploy](/images/docs/appstore/harbor/config_of_harbor_deploy.png)
+
+{{< notice warning >}}
+Don't forget to edit **externalURL**.
+{{</ notice >}}
+
+7. Check the status of deployment, then try to login harbor.
+
+![active_of_harbor](/images/docs/appstore/harbor/active_of_harbor.png)
+
+![overview_of_harbor_login](/images/docs/appstore/harbor/overview_of_harbor_login.png)
+
+### FAQ
+
+1. How to enable http login ?
+
+* set `tls.enabled` as false in step 6.
+* keywords in example yaml
+
+```yaml
+## NOTICE 172.23.5.6 is the test host ip, should use your ip
+expose:
+  type: nodePort
+  tls:
+    enabled: false
+    secretName: ""
+    notarySecretName: ""
+    # commonName should modify
+    commonName: "172.23.5.6"
+  nodePort:
+    # The name of NodePort service
+    name: harbor
+    ports:
+      http:
+        # The service port Harbor listens on when serving with HTTP
+        port: 80
+        # The node port Harbor listens on when serving with HTTP
+        nodePort: 30002
+      https:
+        # The service port Harbor listens on when serving with HTTPS
+        port: 443
+        # The node port Harbor listens on when serving with HTTPS
+        nodePort: 30003
+      # Only needed when notary.enabled is set to true
+      notary:
+        # The service port Notary listens on
+        port: 4443
+        # The node port Notary listens on
+        nodePort: 30004
+
+externalURL: https://172.23.5.6:30003
+
+# The initial password of Harbor admin. Change it from portal after launching Harbor
+harborAdminPassword: "Harbor12345"
+# The secret key used for encryption. Must be a string of 16 chars.
+secretKey: "not-a-secure-key"
+```
+
+2. How to enable https login ?
+
+* set `tls.enabled` as true in step 6, and edit **externalURL** properly.
+* ca certificates is in pod `harbor-core` \'s `/etc/core/ca`.
+* trust the ca certificates by your host first, then restart docker.
```

Comment:
Please write another guide to demonstrate how to create a Secret in a KubeSphere project to connect with Harbor using HTTP and HTTPS respectively.

Reply:
self signed certificates https required restart docker. so we just let user know https is supported, maybe we can redirect this to ks forum.

Reply:
Do not add the external link to the Chinese forum, this is an English guide, you would better to write a step-by-step guide on the website.

Reply:
2 ways to cover:

- HTTPS if user has certificates signed by **public CA**
  this is a simple step, which can bring us big benefit with a little work

- HTTP otherwise
  this is the common scenario, when users don't have formal certs and just want give a try in test environments